packages/metrics/src/Metrics.ts

```diff
@@ -255,25 +263,9 @@ class Metrics extends Utility implements MetricsInterface {
    * @param dimensions - An object with key-value pairs of dimensions
    */
   public addDimensions(dimensions: Dimensions): void {
-    const newDimensions = { ...this.dimensions };
-    for (const dimensionName of Object.keys(dimensions)) {
-      const value = dimensions[dimensionName];
-      if (value) {
-        newDimensions[dimensionName] = value;
-      } else {
-        this.#logger.warn(
-          `The dimension ${dimensionName} doesn't meet the requirements and won't be added. Ensure the dimension name and value are non empty strings`
-        );
-      }
-    }
-    if (Object.keys(newDimensions).length > MAX_DIMENSION_COUNT) {
-      throw new RangeError(
-        `Unable to add ${
-          Object.keys(dimensions).length
-        } dimensions: the number of metric dimensions must be lower than ${MAX_DIMENSION_COUNT}`
-      );
+    for (const [name, value] of Object.entries(dimensions)) {
+      this.addDimension(name, value);
```

Comment:
Just to make sure I understood this change: you are now removing the logic from this method and moving it to `public addDimension(name: string, value: string): void {` because you want to have all this logic in one place, right? With this you iterate through the items and now pass the string values ​​directly to this other method instead of having 2 logics in different places. 
If so, good catch!!

Reply:
yea, that's what it does.